### PR TITLE
HTCONDOR-2614 condor_adstash OpenSearch 2.x compatibility

### PR DESCRIPTION
**Pull request by jasoncpatton**
https://opensciencegrid.atlassian.net/browse/HTCONDOR-2614

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/condor_scripts/adstash/interfaces/opensearch.py [hunk 1]
```diff
@@ -75,3 +92,58 @@ def get_handle(self):
...
+                        updated_mappings[outer_key][inner_key] = mappings[outer_key][inner_key]
+                    update_mappings = True
+        if update_mappings:
+            logging.info(f"Updated mappings for index {index}")
+            logging.debug(f"{pprint.pformat(updated_mappings)}")
+            if OS_VERSION >= (2,0,0):
+                body = {"mappings": updated_mappings}
+                client.indices.put_mapping(index=index, body=body)
+            else:
+                client.indices.put_mapping(index=index, **updated_mappings)
```
Comment (jasoncpatton) [inline src/condor_scripts/adstash/interfaces/opensearch.py:145]: All this just to change these lines 🙃

**Comment by jasoncpatton** [comment]
@iross FYI, GitHub won't let me find you to add as a reviewer for this.

**Comment by timtheisen** [comment]
@jasoncpatton @iross I can fix that for you.

**Comment by timtheisen** [comment]
@iross Invitation sent

**Comment by iross** [comment]
Accepted the invite. Thanks!

File: src/condor_scripts/adstash/interfaces/opensearch.py (hunk 1 above)
Comment (iross) [reply src/condor_scripts/adstash/interfaces/opensearch.py:145]: 😂 never easy, eh?

**Pull request merged**